### PR TITLE
Decouple Shared Components

### PR DESCRIPTION
**Pull request by wey-chiang**
### Background

The Detections Engine Backend in Panther has shared dependencies with PAT. This PR removes the shared code and updates PAT to depend on the new `panther-core` library.

### Changes

* Deleted shared code
* Updated references to point to `panther-core`
* Major version bump

### Testing

* `make ci`

### CODE REVIEW

**Review by darwayne — APPROVED** [review]
🧬 LGTM🧬

**Comment by nmartin413** [comment]
`util.py` had some boto stuff that shouldn't be in `panther_core` - the concept was to keep all networking/remote interactions and clients in `panther_core`'s consumers.

**Comment by wey-chiang** [comment]
> `util.py` had some boto stuff that shouldn't be in `panther_core` - the concept was to keep all networking/remote interactions and clients in `panther_core`'s consumers.

Good callout -- we define our own `get_client` in the detections engine so I'll remove that from `panther-core` and add it back here

**Review by danbiwer — APPROVED** [review]

**Comment by nmartin413** [comment]
@wey-chiang I already removed it in my py exec models ++

**Comment by wey-chiang** [comment]
> @wey-chiang I already removed it in my py exec models ++

Looks like one of my commits reverted those changes heh... it's been removed

**Review by darwayne — DISMISSED** [review]

**Comment by wey-chiang** [comment]
We should hold off on merging and cutting a new release until all the PRs for `panther-analysis` and `panther-enterprise` are ready to go

**Comment by wey-chiang** [comment]
Also including changes for a major version bump in this PR so we're ready to get a release out once this is merged

File: panther_analysis_tool/main.py [hunk 1]
```diff
@@ -1693,7 +1694,7 @@ def setup_parser() -> argparse.ArgumentParser:
         + "managing Panther policies and rules.",
         prog="panther_analysis_tool",
     )
-    parser.add_argument("--version", action="version", version="panther_analysis_tool 0.14.1")
+    parser.add_argument("--version", action="version", version="panther_analysis_tool 1.0.0")
```
Comment (lindsey-w) [inline panther_analysis_tool/main.py:1697]: should we remove the version bump until we are ready to cut a release?

File: setup.py [hunk 2]
```diff
@@ -22,14 +22,14 @@
...
     description=
     'Panther command line interface for writing, testing, and packaging policies/rules.',
     author='Panther Labs Inc',
     author_email='pypi@runpanther.io',
     url='https://github.com/panther-labs/panther_analysis_tool',
-    download_url = 'https://github.com/panther-labs/panther_analysis_tool/archive/v0.14.1.tar.gz',
+    download_url = 'https://github.com/panther-labs/panther_analysis_tool/archive/v1.0.0.tar.gz',
     keywords=['Security', 'CLI'],
     scripts=['bin/panther_analysis_tool'],
     install_requires=install_requires,
```
Comment (lindsey-w) [inline setup.py:35]: do we need to add `panther_core` to the `install_requires` list?

**Review by wey-chiang — COMMENTED** [review]
lazy version bump

**Review by lindsey-w — CHANGES_REQUESTED** [review]
will be scheduling a meeting about deps management

**Pull request closed without merging**